ksql-engine/src/main/java/io/confluent/ksql/function/KsqlUndoableAggregationFunction.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.function;
+
+import org.apache.kafka.connect.data.Schema;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+public abstract class KsqlUndoableAggregationFunction<V, A> extends KsqlAggregateFunction<V, A> {
+  public KsqlUndoableAggregationFunction(
```

Comment:
Perhaps `TableAggregationFunction`?

or just `UndoableAggregationFunction`, i.e., we don't need to name everything `Ksql...` as it makes it a pain when you are looking for classes if everything starts with the same thing. We know it is `Ksql`...